### PR TITLE
Add scouting electron, photon, and muon classes and producers

### PR DESCRIPTION
**Pull request by davidsheffield**
Backport of #11132.

### CODE REVIEW

File: HLTrigger/Egamma/plugins/BuildFile.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<library   file="*.cc" name="HLTriggerEgammaPlugins">
```
Comment (fwyzard) [inline HLTrigger/Egamma/plugins/BuildFile.xml]: HLTrigger/Egamma already defines only plugins (in the src directory), you should not use a new 'plugins' directory.

File: HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc [hunk 2]
```diff
@@ -0,0 +1,282 @@
+// -*- C++ -*-
```
Comment (fwyzard) [inline HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc]: HLTrigger/Egamma already defines only plugins (in the src directory), you should not use a new 'plugins' directory, simply put the .cc file under /src .

File: HLTrigger/Muon/plugins/BuildFile.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<library   file="*.cc" name="HLTriggerMuonPlugins">
```
Comment (fwyzard) [inline HLTrigger/Muon/plugins/BuildFile.xml]: HLTrigger/Muon already defines only plugins (in the /src directory), you should not use a new /plugins directory, just put the .cc files in the /src directory.

File: HLTrigger/Muon/plugins/HLTScoutingMuonProducer.cc [hunk 4]
```diff
@@ -0,0 +1,179 @@
+// -*- C++ -*-
```
Comment (fwyzard) [inline HLTrigger/Muon/plugins/HLTScoutingMuonProducer.cc]: HLTrigger/Muon already defines only plugins (in the /src directory), you should not use a new /plugins directory, just put the .cc files in the /src directory.

**Comment by fwyzard** [comment]
-1

**Comment by davidsheffield** [comment]
@fwyzard I have moved the plugins to the `src` directories.

**Comment by fwyzard** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Pull request merged**